File: Arrays/287. Find the Duplicate Number/solution.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n =len(nums)
        f = [0]*(n+1)
        for i in nums:
            if f[i]==0:
                f[i]+=1
            else:
                return i
        return 0

File: Arrays/287. Find the Duplicate Number/solution.java
import java.util.*;
class Solution {
    public int findDuplicate(int[] nums) {
       int n = nums.length;
       int[] f = new int[n+1];
       for (int i=0;i<n;i++){
        if (f[nums[i]]==0){
            f[nums[i]]+=1;
        }
        else{
            return nums[i];
        }
       }
    return 0;
    }
}